File: 2169-simple-bank-system/simple-bank-system.py
class Bank:

    def __init__(self, balance: List[int]):
        self.balance = balance

    def _is_valid_account(self, account) -> bool:
        if account - 1 < len(self.balance):
            return True
        return False

    def transfer(self, account1: int, account2: int, money: int) -> bool:
        if not self._is_valid_account(account1):
            return False
        if not self._is_valid_account(account2):
            return False
        if not self.withdraw(account1, money):
            return False
        self.deposit(account2, money)
        return True

    def deposit(self, account: int, money: int) -> bool:
        if not self._is_valid_account(account - 1):
            return False
        self.balance[account - 1] += money
        return True

    def withdraw(self, account: int, money: int) -> bool:
        if not self._is_valid_account(account - 1):
            return False
        if self.balance[account - 1] - money < 0:
            return False
        self.balance[account - 1] -= money
        return True


# Your Bank object will be instantiated and called as such:
# obj = Bank(balance)
# param_1 = obj.transfer(account1,account2,money)
# param_2 = obj.deposit(account,money)
# param_3 = obj.withdraw(account,money)

File: 2169-simple-bank-system/simple-bank-system.js
/**
 * @param {number[]} balance
 */
var Bank = function(balance) {
    this.accounts = {}
    for (let i = 0; i < balance.length; i++) {
        this.accounts[i + 1] = balance[i]
    }
};

/** 
 * @param {number} account1 
 * @param {number} account2 
 * @param {number} money
 * @return {boolean}
 */
Bank.prototype.transfer = function(account1, account2, money) {
    if (this.accounts[account1] === undefined) return false
    if (this.accounts[account2] === undefined) return false
    if (this.accounts[account1] < money) return false
    this.accounts[account1] -= money
    this.accounts[account2] += money
    return true
};

/** 
 * @param {number} account 
 * @param {number} money
 * @return {boolean}
 */
Bank.prototype.deposit = function(account, money) {
    if (this.accounts[account] === undefined) return false
    this.accounts[account] += money
    return true
};

/** 
 * @param {number} account 
 * @param {number} money
 * @return {boolean}
 */
Bank.prototype.withdraw = function(account, money) {
    if (this.accounts[account] === undefined) return false
    if (this.accounts[account] < money) return false
    this.accounts[account] -= money
    return true
};

/** 
 * Your Bank object will be instantiated and called as such:
 * var obj = new Bank(balance)
 * var param_1 = obj.transfer(account1,account2,money)
 * var param_2 = obj.deposit(account,money)
 * var param_3 = obj.withdraw(account,money)
 */